ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -160,17 +310,29 @@ status:
     status: "True"
     type: DataObtained
   - lastTransitionTime: "..."
-    message: ""
-    reason: Injected
-    status: "True"
+    message: "some of the targets were not deployed to"
+    reason: PartiallyInjected
+    status: "False"
     type: Deployed
   targets:
-  - namespace: "jdoe-test-ns"
-    secretName: test-remote-secret
-  - namespace: "jdoe-prod-ns"
-    apiUrl: "https://somedomain.copm:443"
-    secretName: test-remote-secret
+  - namespace: "test-target-namespace-1"
+    secretName: secret-from-remote-lsdjf
+    serviceAccountNames:
+    - sa-from-remote-llrkt
+  - namespace: "test-target-namespace-2"
+    secretName: secret-from-remote-lemvs
+    serviceAccountNames:
+    - sa-from-remote-lkejr
+  - namespace: "test-target-namespace-3"
+    secretName: secret-from-remote-kjfdl
+    serviceAccountNames:
+    - sa-from-remote-lmval
+  - namespace: "test-target-namepace-rainbow"
+    apiUrl: "over-the-rainbow"
+    error: "Connection refused"
 ```
+> There are 2 conditions in the status expressing the state of data readiness (`DataObtained` condition type with `AwaitingData` and `DataFound` as possible reasons) and the overall deployment status (`Deployed` condition type with the condition either missing altogether if there are no targets or `PartiallyInjected` or `Injected` reasons).
+> Additionally, the status contains the details of the deployment of each of the targets in the spec. The entries might not come in the same order as in the spec but correspond to each entry in the spec by the `namespace` + `apiUrl` compound key (we don't support 2 targets of a single remote secret pointing to the same namespace atm). The status of the target contains the actual names of the secret and the (optional) service accounts (this is important in case of using `generateName` for the secret or the service account(s)) and optionally also an `error` that explains why certain target was not deployed to.
```

Comment:
IMHO, we just can't support anything else as long as there is a single secret spec for the whole remote secret without the ability to override the name per target. Therefore I wanted to point out that this is a consequence of the model rather than just some arbitrary limitation.